src/gengowatcher/watcher.py

```
@@ -1260,24 +1278,26 @@ def _process_new_job(self, job_id, title, reward, url, source, source_meta=None)
+                    and self.cancellation_manager.should_cancel_for_job(
+                        float(job_data.get("reward", 0.0)), str(job_data.get("id"))
+                    )
+                ):
+                    self.logger.info(
+                        "Better opportunity detected; cancelling current job before acceptance"
+                    )
+                    cancellation_future = self._cancellation_executor.submit(
+                        self._async_cancel_current_job_wrapper, job_data
+                    )
+                    cancellation_future.result()
+            except Exception as e:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)